Controllers/Frontend/Adyen.php

```diff
@@ -116,31 +101,26 @@ public function resetValidPaymentSessionAction()
     /**
      * @throws AdyenException
      */
-    private function paymentDetails(string $post, string $detail)
+    public function ajaxThreeDsAction()
```

Comment:
I believe that there's a way to keep this function generically as paymentDetails? What do you think @cyattilakiss

Reply:
@acampos1916 yes indeed you are right. It's better to keep it as paymentDetails than specifiecally assigning it to threeDsActions
All the actions (including 3DS( can be handled via the paymentDetails endpoint so it makes sense to keep only 1 endpoint

Reply:
Right. @bortefi do you need the 3DS2 result specifically for some frontend logic? If not the response from /payments/details could be handled generically

Reply:
In the previous version "paymentDetails" only handled 3DS alias actions hence the renaming to one general 3DS related method name.

It's a callback url internally used, renaming this to `ajaxPaymentDetails` or `handlePaymentDetails` is fine.
Front-end only needs to be updated with the new action Method for url callback of the 3DS handling.

Reply:
Ah you're right @bortefi, I took a closer look and see it better now. This was being used by the ChallengeShopper and IdentifyShopper endpoints and only for the 3DS2 flow, and you're making it directly available now.

We still want to make it generic so that it can be reused by other flows if needed, e.g. when implementing a custom checkout flow that needs the paymentsDetails validation for more payment methods. And the threeDSResult empty check is not needed. Could you please refactor it to make it generic?

Reply:
See refeactored code below: https://github.com/Adyen/adyen-shopware5/pull/155#discussion_r615727958_